Not compatible with Tornado 5

Hello,

I tried with the last master to launch a small executable like in Step by Step install
However I keep getting this error:

```
2018-03-14 22:17:45 circus[14844] [INFO] Starting master on pid 14844
2018-03-14 22:17:45 circus[14844] [INFO] sockets started
2018-03-14 22:17:45 circus[14844] [ERROR] exception __init__() takes 3 positional arguments but 4 were given caught
```

My circus.ini file is below

```
[circus]
statsd = 1
httpd = 1

[watcher:server]
cmd = "C:\server.exe"
```
The exception seem to be catched so I cannot say where is the error.
I run on Windows 7 64 bits with python 3.6.3

Do you have any hint from were the error come from?

Regards

Same thing here

```
2018-03-18 03:43:47 circus[147] [INFO] Starting master on pid 147
2018-03-18 03:43:47 circus[147] [ERROR] exception __init__() takes 3 positional arguments but 4 were given caught
```

on debian jessie

**same as me.**
centos 7.2, tornado 5.0.1 circus 0.14.0
python 3.6.3
error msg:
circusd /etc/circus/circus.ini
2018-03-20 18:22:45 circus[3779] [INFO] Starting master on pid 3779
2018-03-20 18:22:45 circus[3779] [INFO] sockets started
2018-03-20 18:22:45 circus[3779] [ERROR] exception __init__() takes 3 positional arguments but 4 were given caught

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,8 +9,14 @@ addons:
 before_install:
   # Try https://github.com/travis-ci/travis-ci/issues/8363#issuecomment-327857631
   # to workaround Travis having broken Python3.5, and flaky 3.4
-  - pyenv global system $TRAVIS_PYTHON_VERSION
-
+  #
+  #
+  - pyenv global ${TRAVIS_PYTHON_VERSION}
+  - >
+    if [[ $TRAVIS_PYTHON_VERSION = "3.5" ]]; then deactivate; sudo rm -rf /opt/python /home/travis/virtualenv; sudo add-apt-repository -y ppa:deadsnakes/ppa;
+    sudo apt-get update; sudo apt-get install -y python3.5 python3.5-dev; pyenv rehash; pyenv global system; pip install -U virtualenv pip;
+    virtualenv --python=/usr/bin/python3.5 --no-site-packages ~/working_venv; source ~/working_venv/bin/activate;
+    python --version; which python; pip --version; fi
 before_script:
   # Add an IPv6 config - see the corresponding Travis issue
   # https://github.com/travis-ci/travis-ci/issues/8361
@@ -41,7 +47,7 @@ jobs:
       env: TOX_ENV=py26-no-gevent
 
 script:
-    - tox -e $TOX_ENV
+    - tox -v -e $TOX_ENV
 
 install:
     - pip install tox

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,24 +1,48 @@
 language: python
 sudo: false
-python: 3.5
 addons:
   apt:
     packages:
       - libev-dev
       - libevent-dev
-env:
-  - TOX_ENV=py26
-  - TOX_ENV=py26-no-gevent
-  - TOX_ENV=py27
-  - TOX_ENV=py27-no-gevent
-  - TOX_ENV=py33
-  - TOX_ENV=py34
-  - TOX_ENV=py35
-  - TOX_ENV=docs
-  - TOX_ENV=flake8
+
+before_install:
+  # Try https://github.com/travis-ci/travis-ci/issues/8363#issuecomment-327857631
+  # to workaround Travis having broken Python3.5, and flaky 3.4
+  - pyenv global system $TRAVIS_PYTHON_VERSION
+
+before_script:
+  # Add an IPv6 config - see the corresponding Travis issue
+  # https://github.com/travis-ci/travis-ci/issues/8361
+  # Taken from https://github.com/travis-ci/travis-ci/issues/8361#issuecomment-350497804
+  - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then 
+      sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6';
+    fi
+
+jobs:
+  include:
+    - python: 3.5
+      env: TOX_ENV=py35
+    - python: 3.5
+      env: TOX_ENV=docs
+    - python: 3.5
+      env: TOX_ENV=flake8
+    - python: 3.4
+      env: TOX_ENV=py34
+    - python: 3.3
+      env: TOX_ENV=py33
+    - python: 2.7
+      env: TOX_ENV=py27
+    - python: 2.7
+      env: TOX_ENV=py27-no-gevent
+    - python: 2.6
+      env: TOX_ENV=py26
+    - python: 2.6
+      env: TOX_ENV=py26-no-gevent
 
 script:
     - tox -e $TOX_ENV
+
 install:
     - pip install tox
 

--- a/setup.py
+++ b/setup.py
@@ -6,7 +6,7 @@ if not hasattr(sys, 'version_info') or sys.version_info < (2, 6, 0, 'final'):
     raise SystemExit("Circus requires Python 2.6 or higher.")
 
 
-install_requires = ['psutil', 'pyzmq>=13.1.0', 'tornado>=3.0,<5.0', 'six']
+install_requires = ['psutil', 'pyzmq>=13.1.0,<17.0', 'tornado>=3.0,<5.0', 'six']
 
 try:
     import argparse     # NOQA

--- a/setup.py
+++ b/setup.py
@@ -6,7 +6,7 @@ if not hasattr(sys, 'version_info') or sys.version_info < (2, 6, 0, 'final'):
     raise SystemExit("Circus requires Python 2.6 or higher.")
 
 
-install_requires = ['psutil', 'pyzmq>=13.1.0', 'tornado>=3.0', 'six']
+install_requires = ['psutil', 'pyzmq>=13.1.0', 'tornado>=3.0,<5.0', 'six']
 
 try:
     import argparse     # NOQA

--- a/tox.ini
+++ b/tox.ini
@@ -23,6 +23,7 @@ deps =
     gevent
     circus-web
     tornado>=3.0,<5.0
+    pyzmq>=13.1.0,<17.0
 
 commands =
     nosetests -vs --with-coverage --cover-package=circus circus/tests

--- a/tox.ini
+++ b/tox.ini
@@ -1,60 +1,59 @@
-language: python
-sudo: false
-addons:
-  apt:
-    packages:
-      - libev-dev
-      - libevent-dev
+[tox]
+envlist = py26,py26-no-gevent,py27,py27-no-gevent,py33,py34,py35,flake8,docs
 
-before_install:
-  # Try https://github.com/travis-ci/travis-ci/issues/8363#issuecomment-327857631
-  # to workaround Travis having broken Python3.5, and flaky 3.4
-  #
-  #
-  - pyenv global ${TRAVIS_PYTHON_VERSION}
-  - >
-    if [[ $TRAVIS_PYTHON_VERSION = "3.5" ]]; then deactivate; sudo rm -rf /opt/python /home/travis/virtualenv; sudo add-apt-repository -y ppa:deadsnakes/ppa;
-    sudo apt-get update; sudo apt-get install -y python3.5 python3.5-dev; pyenv rehash; pyenv global system; pip install -U virtualenv pip;
-    virtualenv --python=/usr/bin/python3.5 --no-site-packages ~/working_venv; source ~/working_venv/bin/activate; 
-    python --version; which python; pip --version; fi
-before_script:
-  # Add an IPv6 config - see the corresponding Travis issue
-  # https://github.com/travis-ci/travis-ci/issues/8361
-  # Taken from https://github.com/travis-ci/travis-ci/issues/8361#issuecomment-350497804
-  - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then 
-      sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6';
-    fi
+[testenv:py26]
+deps =
+    {[testenv]deps}
+    gevent<1.2
+    unittest2
 
-jobs:
-  include:
-    - python: 3.6
-      env: TOX_ENV=py36
-    - python: 3.5
-      env: TOX_ENV=py35
-    - python: 3.5
-      env: TOX_ENV=docs
-    - python: 3.5
-      env: TOX_ENV=flake8
-    - python: 3.4
-      env: TOX_ENV=py34
-    - python: 3.3
-      env: TOX_ENV=py33
-    - python: 2.7
-      env: TOX_ENV=py27
-    - python: 2.7
-      env: TOX_ENV=py27-no-gevent
-    - python: 2.6
-      env: TOX_ENV=py26
-    - python: 2.6
-      env: TOX_ENV=py26-no-gevent
+[testenv:py26-no-gevent]
+deps =
+    {[testenv]deps}
+    unittest2
 
-script:
-    - tox -v -e $TOX_ENV
+[testenv:py27]
+passenv = PWD TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
 
-install:
-    - pip install tox
+deps =
+    {[testenv]deps}
+    nose-cov
+    coverage
+    coveralls
+    gevent
+    circus-web
 
-notifications:
-    email: tarek@mozilla.com
-    irc: "irc.freenode.org#mozilla-circus"
-    on_success: change
+commands =
+    nosetests -vs --with-coverage --cover-package=circus circus/tests
+    coveralls
+
+[testenv]
+passenv = PWD
+deps =
+    nose
+    mock
+    PyYAML
+    six
+    tornado>=3.0,<5.0
+    pyzmq>=13.1.0,<17.0
+
+setenv =
+    TESTING=1
+    PYTHONHASHSEED=random
+    PYTHONUNBUFFERED=1
+
+commands =
+    nosetests -vs circus/tests
+
+
+[testenv:docs]
+whitelist_externals = make
+deps =
+    sphinx
+    mozilla-sphinx-theme
+commands = make -C docs html
+
+
+[testenv:flake8]
+deps = flake8==2.1.0
+commands = flake8 circus

--- a/tox.ini
+++ b/tox.ini
@@ -22,6 +22,7 @@ deps =
     coveralls
     gevent
     circus-web
+    tornado>=3.0,<5.0
 
 commands =
     nosetests -vs --with-coverage --cover-package=circus circus/tests

--- a/tox.ini
+++ b/tox.ini
@@ -1,59 +1,60 @@
-[tox]
-envlist = py26,py26-no-gevent,py27,py27-no-gevent,py33,py34,py35,flake8,docs
+language: python
+sudo: false
+addons:
+  apt:
+    packages:
+      - libev-dev
+      - libevent-dev
 
-[testenv:py26]
-deps =
-    {[testenv]deps}
-    gevent<1.2
-    unittest2
+before_install:
+  # Try https://github.com/travis-ci/travis-ci/issues/8363#issuecomment-327857631
+  # to workaround Travis having broken Python3.5, and flaky 3.4
+  #
+  #
+  - pyenv global ${TRAVIS_PYTHON_VERSION}
+  - >
+    if [[ $TRAVIS_PYTHON_VERSION = "3.5" ]]; then deactivate; sudo rm -rf /opt/python /home/travis/virtualenv; sudo add-apt-repository -y ppa:deadsnakes/ppa;
+    sudo apt-get update; sudo apt-get install -y python3.5 python3.5-dev; pyenv rehash; pyenv global system; pip install -U virtualenv pip;
+    virtualenv --python=/usr/bin/python3.5 --no-site-packages ~/working_venv; source ~/working_venv/bin/activate; 
+    python --version; which python; pip --version; fi
+before_script:
+  # Add an IPv6 config - see the corresponding Travis issue
+  # https://github.com/travis-ci/travis-ci/issues/8361
+  # Taken from https://github.com/travis-ci/travis-ci/issues/8361#issuecomment-350497804
+  - if [ "${TRAVIS_OS_NAME}" == "linux" ]; then 
+      sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6';
+    fi
 
-[testenv:py26-no-gevent]
-deps =
-    {[testenv]deps}
-    unittest2
+jobs:
+  include:
+    - python: 3.6
+      env: TOX_ENV=py36
+    - python: 3.5
+      env: TOX_ENV=py35
+    - python: 3.5
+      env: TOX_ENV=docs
+    - python: 3.5
+      env: TOX_ENV=flake8
+    - python: 3.4
+      env: TOX_ENV=py34
+    - python: 3.3
+      env: TOX_ENV=py33
+    - python: 2.7
+      env: TOX_ENV=py27
+    - python: 2.7
+      env: TOX_ENV=py27-no-gevent
+    - python: 2.6
+      env: TOX_ENV=py26
+    - python: 2.6
+      env: TOX_ENV=py26-no-gevent
 
-[testenv:py27]
-passenv = PWD TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
+script:
+    - tox -v -e $TOX_ENV
 
-deps =
-    {[testenv]deps}
-    nose-cov
-    coverage
-    coveralls
-    gevent
-    circus-web
+install:
+    - pip install tox
 
-commands =
-    nosetests -vs --with-coverage --cover-package=circus circus/tests
-    coveralls
-
-[testenv]
-passenv = PWD
-deps =
-    nose
-    mock
-    PyYAML
-    six
-    tornado>=3.0,<5.0
-    pyzmq>=13.1.0,<17.0
-
-setenv =
-    TESTING=1
-    PYTHONHASHSEED=random
-    PYTHONUNBUFFERED=1
-
-commands =
-    nosetests -vs circus/tests
-
-
-[testenv:docs]
-whitelist_externals = make
-deps =
-    sphinx
-    mozilla-sphinx-theme
-commands = make -C docs html
-
-
-[testenv:flake8]
-deps = flake8==2.1.0
-commands = flake8 circus
+notifications:
+    email: tarek@mozilla.com
+    irc: "irc.freenode.org#mozilla-circus"
+    on_success: change

--- a/tox.ini
+++ b/tox.ini
@@ -22,8 +22,6 @@ deps =
     coveralls
     gevent
     circus-web
-    tornado>=3.0,<5.0
-    pyzmq>=13.1.0,<17.0
 
 commands =
     nosetests -vs --with-coverage --cover-package=circus circus/tests
@@ -36,6 +34,8 @@ deps =
     mock
     PyYAML
     six
+    tornado>=3.0,<5.0
+    pyzmq>=13.1.0,<17.0
 
 setenv =
     TESTING=1

